numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -387,14 +392,18 @@ array_implement_c_array_function_creation(
         return Py_NotImplemented;
     }
 
-    PyObject *like_arg = PyDict_GetItem(kwargs, npy_ma_str_like);
+    PyObject *like_arg = PyDict_GetItemWithError(kwargs, npy_ma_str_like);
     if (like_arg == NULL) {
         return NULL;
     }
-    else if (!get_array_function(like_arg)) {
-        return PyErr_Format(PyExc_TypeError,
-                "The `like` argument must be an array-like that implements "
-                "the `__array_function__` protocol.");
+    else {
+        PyObject *tmp_has_override = get_array_function(like_arg);
+        if (tmp_has_override == NULL) {
+            return PyErr_Format(PyExc_TypeError,
+                    "The `like` argument must be an array-like that "
+                    "implements the `__array_function__` protocol.");
+        }
+        Py_DECREF(tmp_has_override);
```

Comment:
This is the larger difference. The code path looks different on `main`.  This is much like the diff just above.